packages/react-dom/src/events/ReactDOMEventListener.js

```diff
@@ -268,7 +261,14 @@ function dispatchDiscreteEvent(
   container,
   nativeEvent,
 ) {
-  flushDiscreteUpdatesIfNeeded(nativeEvent.timeStamp);
+  if (
+    !enableLegacyFBSupport ||
+    // If we have Legacy FB support, it means we've already
+    // flushed for this event and we don't need to do it again.
+    (enableLegacyFBSupport && (eventSystemFlags & LEGACY_FB_SUPPORT) === 0)
```

Comment:
`enableLegacyFBSupport &&` is superfluous here. The previous condition already excluded it.